src/Core/Services/CipherService.cs

```diff
@@ -1286,6 +1286,34 @@ async Task encryptAndAddHistoryAsync(PasswordHistoryView model, PasswordHistory
             cipher.PasswordHistory = encPhs;
         }
 
+        public async Task<string> CreateNewLoginForPasskeyAsync(string rpId)
+        {
+            var newCipher = new CipherView
+            {
+                Name = rpId,
+                Type = CipherType.Login,
+                Login = new LoginView
+                {
+                    Uris = new List<LoginUriView>
+                    {
+                        new LoginUriView { Uri = rpId }
+                    }
+                },
+                Card = new CardView(),
+                Identity = new IdentityView(),
+                SecureNote = new SecureNoteView
+                {
+                    Type = SecureNoteType.Generic
+                },
+                Reprompt = CipherRepromptType.None
+            };
```

Comment:
issue: can we also assign the username here to align with [PM-4661 Populate login item Username with passkey username](https://bitwarden.atlassian.net/browse/PM-4661)

Reply:
I've added that [here](https://github.com/bitwarden/mobile/pull/3019/files#diff-2c6c72919b26d3ded8ed10ff92d929050a8b8087adffee3d5dff8b896d260b5bR52) when creating a login with the Add button but forgot about adding it to this method, will add that as part of that PR.